dns-server/tests/basic_test.rs

```diff
@@ -571,35 +571,23 @@ async fn dns_records_list(
         .unwrap_or_else(HashMap::new))
 }
 
-/// issue a DNS query of `record_ty` records for `name`.
+/// Issue a DNS query of `record_ty` records for `name`.
 ///
-/// in most tests we just use `trust-dns-resolver` for a higher-level interface
-/// to making DNS queries and handling responses. this is also the crate used
-/// elsewhere to handle DNS responses in Omicron. however, it is slightly
+/// In most tests we just use `trust-dns-resolver` for a higher-level interface
+/// to making DNS queries and handling responses. This is also the crate used
+/// elsewhere to handle DNS responses in Omicron. However, it is slightly
 /// higher-level than the actual wire response we produce from the DNS server in
 /// this same crate; to assert on responses we send on the wire, this issues a
 /// query using `trust-dns-client` and returns the corresponding `DnsResponse`.
-// FIXME(ixi): while `trust-dns-client` has an `AsyncClient` which theoretically
-// integrates nicely with tokio use in this test, I can't figure out how to
-// instantiate it! `SyncClient` is just as good but makes for a slightly messier
-// test.
 async fn raw_dns_client_query(
     resolver_addr: std::net::SocketAddr,
     name: Name,
     record_ty: RecordType,
 ) -> Result<DnsResponse, ClientError> {
-    let dns_client_conn = UdpClientConnection::new(resolver_addr)
-        .expect("can prepare UdpClientConnection");
-    let trust_client = SyncClient::new(dns_client_conn);
-
-    tokio::task::spawn_blocking(move || {
-        // query for an A record. the answer we expect is:
-        // * no error: the domain exists, so NXDOMAIN would be wrong
-        // * no answers: we ask specifically for a record type the server does not have
-        // * no additionals: the server could return AAAA records as additionals
-        //   to an A query, but does not currently.
-        trust_client.query(&name, DNSClass::IN, record_ty)
-    })
-    .await
-    .expect("query and test is ok")
+    let stream = UdpClientStream::<tokio::net::UdpSocket>::new(resolver_addr);
+    let (mut trust_client, bg) = AsyncClient::connect(stream).await.unwrap();
+
+    tokio::spawn(bg);
```

Comment:
Should we cancel and and then wait for this after we do the query?

Reply:
i don't think we'd get anything for the effort to do that: `bg` *does* need to be polled, since that's what drives `trust_client`'s I/O. so if we held `bg` and cancelled it after ` trust_client.query()`, i'm pretty sure the query would just time out .. :D

so we'd need to thread some signal through that the query is done and we should tear down `bg`. but dropping `trust_client` has a really similar outcome in getting `DnsExchangeBackground` shutdown [here](https://github.com/hickory-dns/hickory-dns/blob/d07ef69/crates/proto/src/xfer/dns_exchange.rs#L220-L222).

is there state you're thinking might persist if `bg` were to leak?

Reply:
I'm just worried about leaking those tasks (and `bg` itself) every time we run a query, plus general principles -- it makes me nervous when stuff kicks stuff off in the background and forgets about it.

Reply:
yeah, fair. i'm less concerned about that here because this is only used in the context of some individual test, and the whole executor gets torn down when the test finishes?